src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -17,6 +18,34 @@
   private WPI_TalonSRX right_master_talon;
   private WPI_TalonSRX right_slave_talon;
   private MecanumDrive drive;
+  int P, I, D = 1;
+  int integral, previous_error, setpoint = 0;
+  Gyro gyro;
+  DifferentialDrive robotDrive;
+  private double rcw;
+
+
+  public void Drive(Gyro gyro) {
+      this.gyro = gyro;
+  }
+
+  public void setSetpoint(int setpoint)
+  {
+      this.setpoint = setpoint;
+  }
+
+  public void PID(){
+      double error = setpoint - gyro.getAngle(); // Error = Target - Actual
+      this.integral += (error*.02); // Integral is increased by the error*time (which is .02 seconds using normal IterativeRobot)
+      double derivative = (error - this.previous_error) / .02;
+      this.rcw = P*error + I*this.integral + D*derivative;
+  }
```

Comment:
This is not at all what I meant, what I wanted you to do was to add the drivetrain PID constants to constants.java, and then use talon.config_kP and the other config commands to give the constants to the talons.